HW15/HomeWork15.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <algorithm>
+
+struct Student
+{
+	std::string Name;
+	int Marks[4];
+};
+
+// Task 1
+
+int GetAverageMark(const Student& SomeStudent)
+{
+	int Sum = 0;
+	for (int i = 0; i < sizeof(SomeStudent.Marks) / sizeof(int); i++)
+	{
+		Sum += SomeStudent.Marks[i];
+	}
+	return Sum / (sizeof(SomeStudent.Marks) / sizeof(int));
+}
+
+// Task 3
+
+bool CompareByAvgMark(const Student& Student1, const Student& Student2)
+{
+	return GetAverageMark(Student1) > GetAverageMark(Student2);
+}
+
+void SortStudentsByAvg(Student* Students, int Size)
+{
+	std::sort(Students, Students + Size , CompareByAvgMark);
+}
+
+// Task 4
+
+Student* GetBestStudent(Student* Students, int Size)
+{
+	Student* BestStudent = Students;
+	for (int i = 1; i < Size ; i++)
+	{
+		if (GetAverageMark(*BestStudent) < GetAverageMark(Students[i]))
+		{
+			BestStudent = &Students[i];
+		}
+	}
+	return BestStudent;
```

Comment:
So, let's imagine the best student is at position 0. In that case you algo will calculate average N times for the same student without need